app/components/dashboard/dashboard-controller.ts

```diff
@@ -0,0 +1,308 @@
+import {VolumeService} from '../rest/volume';
+import {PoolService} from '../rest/pool';
+
+declare var require : any;
+var numeral = require("numeral");
+
+export class DashboardController {
+    private self = this;
+    private config: any;
+    private clusters: Array<any>;
+    private clustersWarning: Array<any>;
+    private clustersCritical: Array<any>;
+    private hosts: Array<any>;
+    private hostsWarning: Array<any>;
+    private hostsCritical: Array<any>;
+    private volumes: Array<any>;
+    private volumesWarning: Array<any>;
+    private volumesCritical: Array<any>;
+    private pools: Array<any>;
+    private poolsWarning: Array<any>;
+    private poolsCritical: Array<any>;
+    private services: Array<any>;
+    private servicesWarning: Array<any>;
+    private servicesCritical: Array<any>;
+    
+    private clusterTypes: Array<any>;
+    private storageTiers: Array<any>;
+    private totalCapacity: any;
+    private trendCapacity: any;
+
+
+
+
+
+    static $inject: Array<string> = [
+        '$scope',
+        '$location',
+        '$log',
+        'ClusterService',
+        'ServerService',
+        'VolumeService',
+        'PoolService',
+    ];
+
+    constructor(private $scope: ng.IScope,
+        private $location: ng.ILocationService,
+        private $log: ng.ILogService,
+        private clusterService: ClusterService,
+        private serverService: ServerService,
+        private volumeService: VolumeService,
+        private poolService: PoolService) {
+
+
+         this.config = { capacityByType: true, capacityByTier: false };
+         this.clusterTypes = [
+            { id:1, name: 'Block', color: '#48b3ea', type: 'donut' },
+            { id:2, name: 'File', color: '#0088ce' ,type: 'donut' },
+            { id:3, name: 'Object', color: '#00659c', type: 'donut' },
+            { id:9, name: 'Free', color: '#969696', type: 'donut' } ];
+         this.storageTiers = [
+            { id:1, name: 'Default', color: '#48b3ea', type: 'donut' },
+            { id:2, name: 'Faster', color: '#0088ce', type: 'donut' },
+            { id:3, name: 'Slower', color: '#00659c', type: 'donut' },
+            { id:9, name: 'Free', color: '#969696', type: 'donut' } ];
+         this.totalCapacity = {
+                freeGB: 0, usedGB: 0, totalGB: 0,
+                freeFormatted: '0 B', usedFormatted: '0 B', totalFormatted: '0 B'
+            };
+
+         this.totalCapacity.legends = this.clusterTypes;
+         this.totalCapacity.values = [];
+         this.totalCapacity.byType = [];
+         this.totalCapacity.byTier = [];
+         this.services = [];
+
+         this.trendCapacity = {};
+         this.trendCapacity.legends = [
+                { id:1, name: 'Used', color: '#39a5dc', type: 'area-spline' }
+         ];
+         this.trendCapacity.values = [];
+         this.trendCapacity.selected = { used: 0, isTotal: true, type: '' };
+
+         
+         this.serverService.getList().then(this.updateHostData);
```

Comment:
This can be written as 
```
this.serverService.getList().then((clusters) => this.updateHostData(clusters));
```

So the below function can be written as 
```
public updateClusterData(clusters: any) {
 ...
}
```

This is mainly to improve the readability.

The same can be done for all 4 functions